src/source/tile.js

```diff
@@ -595,7 +595,7 @@ class Tile {
     setTexture(img: TextureImage, painter: Painter) {
         const context = painter.context;
         const gl = context.gl;
-        this.texture = painter.getTileTexture(img.width);
+        this.texture = this.texture || painter.getTileTexture(img.width);
```

Comment:
This is an optimization that reduces the number of `new Texture` calls, but I'm not entirely sure that this is a legal thing to do here.

Reply:
@ansis @mourner, do you see any potential issues with this?

Reply:
Caching is supposed to be handled by `getTileTexture`, and new texture should be created only when it's a new image size we have no cached texture for. So it looks a little fishy here, needs further debugging...

Reply:
This looks ok to me. Previously we assumed setTexture would only be called once per tile. Now it might be called multiple times to update the texture. If we already have a texture for that tile, reusing that one should be what we want